cliconfig/config.go

```diff
@@ -178,7 +189,7 @@ func Load(configDir string) (*ConfigFile, error) {
 	}
 
 	// Can't find latest config file so check for the old one
-	confFile := filepath.Join(homedir.Get(), oldConfigfile)
+	confFile := getDefaultConfigDir(oldConfigfile)
```

Comment:
@runcom I was thinking to completely deprecate the `.dockercfg` file, so to not change the behavior for that one. Just searched when we moved to the new `.docker` directory, and that was in https://github.com/docker/docker/pull/12009, which was introduced in docker 1.7.0